### PR TITLE
website: update next and nextjs-scripts

### PR DESCRIPTION
**Pull request by BRKalow**
Upgrading next to v10 and nextjs-scripts to v15.

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -16,8 +16,8 @@
     "@hashicorp/react-section-header": "3.0.1",
     "@hashicorp/react-subnav": "7.1.0",
     "@hashicorp/react-vertical-text-block-list": "3.0.1",
-    "next": "9.5.5",
-    "next-mdx-remote": "^1.0.1",
+    "next": "10.0.6",
+    "next-mdx-remote": "^2.1.3",
```
Comment (BRKalow) [inline website/package.json:20]:
```suggestion
    "next-mdx-remote": "2.1.3",
```

File: website/next.config.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ console.log(`Environment: ${process.env.HASHI_ENV}`)
 
 module.exports = withHashicorp({
   defaultLayout: true,
-  transpileModules: ['is-absolute-url'],
+  transpileModules: ['is-absolute-url', '@hashicorp/react-image'],
```
Comment (jescalan) [inline website/next.config.js:8]: Confused about this one, don't we transpile all the global components by default now?

**Review by jescalan — APPROVED** [review]

File: website/next.config.js (hunk 2 above)
Comment (BRKalow) [reply website/next.config.js:8]: We should, but this dependency was nested so it didn't get picked up by my solution.  I'll add a ticket to handle this case within scripts.

Comment (BRKalow) [reply website/next.config.js:8]: Ticket: https://app.asana.com/0/1100423001970639/1199916284212909/f

**Pull request merged**